source/common/upstream/eds.cc

```diff
@@ -63,33 +59,28 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources, const std::
     throw EnvoyException(fmt::format("Unexpected EDS cluster (expecting {}): {}", cluster_name_,
                                      cluster_load_assignment.cluster_name()));
   }
+
+  auto& priority_state = priority_state_manager_->priority_state_;
+  priority_state.clear();
```

Comment:
Isn't clearing and creating every time going to be more computationally expensive?  It'd be good if we can do this refactor with minimal performance regression

Reply:
Does it makes sense to make PriorityStateManager a local variable in onConfigUpdate? It would mean reverting the `Config::Utility::checkLocalInfo` call, but other than that check, PSM isn't needed beyond the scope on onConfigUpdate.